src/Common/src/Interop/Unix/libc/Interop.getcwd.cs

```diff
@@ -10,7 +10,57 @@ internal static partial class Interop
 {
     internal static partial class libc
     {
+        internal static unsafe string getcwd()
+        {
+            const int StackLimit = 256;
+
+            // First try to get the path into a buffer on the stack
+            byte* stackBuf = stackalloc byte[StackLimit];
+            string result = getcwd(stackBuf, StackLimit);
+            if (result != null)
+                return result;
+
+            // If that was too small, try increasing large buffer sizes
+            // until we get one that works or until we hit MaxPath.
+            int maxPath = Interop.libc.MaxPath;
+            int bufferSize = StackLimit;
+            while ((bufferSize = Math.Min(bufferSize * 2, maxPath)) < maxPath)
```

Comment:
Also, note that your queried maxPath value can be -1 in situations where a system describes itself as having no inherent limit.  I don't know that it's anything that would really be encountered in the wild, but the manpage describes it that way.